### PR TITLE
New: MediaBrowser notifications

### PR DESCRIPTION
**Pull request by markus101**
@kayone @Taloth 

Converted it from RestSharp to our own HttpClient and since MediaBrowser actually supports library updates now we can actually release it.

### CODE REVIEW

File: src/NzbDrone.Core/Notifications/MediaBrowser/MediaBrowser.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using FluentValidation.Results;
+using NzbDrone.Common;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Tv;
+
+namespace NzbDrone.Core.Notifications.MediaBrowser
+{
+    public class MediaBrowser : NotificationBase<MediaBrowserSettings>
```
Comment (markus101) [inline src/NzbDrone.Core/Notifications/MediaBrowser/MediaBrowser.cs]: Any particular reason why? Currently we use the class name as the name when selection the type, so this would need to change or look bad, but I don't think re-factoring all that is warranted currently.

File: src/NzbDrone.Core/Notifications/MediaBrowser/MediaBrowserService.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            {
+                if (ex.Response.StatusCode == HttpStatusCode.Unauthorized)
+                {
+                    return new ValidationFailure("ApiKey", "API Key is incorrect");
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.ErrorException("Unable to send test message: " + ex.Message, ex);
+                return new ValidationFailure("Host", "Unable to send test message");
```
Comment (markus101) [inline src/NzbDrone.Core/Notifications/MediaBrowser/MediaBrowserService.cs]: Agreed, I'll send back the exception message.

File: src/NzbDrone.Core/Notifications/MediaBrowser/MediaBrowserProxy.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            _logger.Trace("Response: {0}", response.Content);
+
+            CheckForError(response);
+
+            return response.Content;
+        }
+
+        private HttpRequest BuildRequest(string path, MediaBrowserSettings settings)
+        {
+            var url = String.Format(@"http://{0}/mediabrowser", settings.Address);
```
Comment (markus101) [inline src/NzbDrone.Core/Notifications/MediaBrowser/MediaBrowserProxy.cs]: You need a valid URI to build the request, and need a request to build the segment. Chicken vs Egg.

File: src/NzbDrone.Core/Notifications/MediaBrowser/MediaBrowser.cs (hunk 1 above)
Comment (kayone) [inline src/NzbDrone.Core/Notifications/MediaBrowser/MediaBrowser.cs]: It's just too generic. Honestly I don't like the `NzbGet`, `Sabnzbd` classes either.

Comment (markus101) [inline src/NzbDrone.Core/Notifications/MediaBrowser/MediaBrowser.cs]: Fair enough. This requires some other architectural changes, which I agree is a good idea (added to Trello: https://trello.com/c/2AW7y73n/912-be-more-explicit-with-class-names-for-3rd-party-integrations).

I'd like to get this merged in and we can make those other changes in another PR.
